src/freelist.rs

```diff
@@ -9,13 +17,12 @@ use crate::{ffi, AsPyPointer, FromPyPointer, PyAny, Python};
 use std::mem;
 use std::os::raw::c_void;
 
-/// Implementing this trait for custom class adds free allocation list to class.
-/// The performance improvement applies to types that are often created and deleted in a row,
-/// so that they can benefit from a freelist.
+/// Implements a freelist. Using `#[pyclass(freelist = N)]` on a Rust struct will implement this trait.
```

Comment:
It's perhaps worth adding a note that it's really not expected for the user to ever interact with this trait. As far as I'm concerned, it's an implementation detail that's very likely to need to be rewritten when we optimize pyclass creation. (We'll preserve `#[pyclass(freelist = N)]` but not necessarily any of the mechanics.)

Reply:
Done.